onnxruntime/test/wasm/test_inference.cc

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <gtest/gtest.h>
+
+#include "core/session/onnxruntime_cxx_api.h"
```

Comment:
This test works but it is very tricky.  When you do static linking, a lot of things changes. For example, we carefully avoid using the public C++ APIs because they all depend on a global initializer.  When it initializes, you don't know if the other parts of ORT are already initialized. You can only assume they are not. Then, how would you know it still works as expected? Also, C ABI make no value in static linking.  If a program static link to ORT, it should use the internal C++ APIs like onnxruntime::InferenceSession. But they are not well documented yet.